File: 200/240 - Search a 2D Matrix II.c
bool searchMatrix(int** matrix, int matrixRowSize, int matrixColSize, int target) {
    int r = matrixRowSize - 1, c = 0;
    
    if (! (matrixRowSize && matrixColSize))
        return 0;
    
    while (1) {
        if (matrix[r][c] < target)
            ++c;
        else if (matrix[r][c] == target)
            return 1;
        else
            --r;
        
        if (r < 0 || c >= matrixColSize)
            break;
    }
    return 0;
}

File: csharp/240.search-a-2-d-matrix-ii.cs
/*
 * @lc app=leetcode id=240 lang=csharp
 *
 * [240] Search a 2D Matrix II
 */
public class Solution {
    public bool SearchMatrix(int[,] matrix, int target) {
        int m = matrix.GetLength(0), n = matrix.GetLength(1), r = m - 1, c = 0;
        while (r >= 0 && c < n) {
            if (matrix[r, c] > target) {
                --r;
            } else if (matrix[r, c] == target) {
                return true;
            } else {
                ++c;
            }
        }
        return false;
    }
}

